### PR TITLE
Comment to explain protobuf version pinning.

### PR DESCRIPTION
**Pull request by hjoliver**
This is a small change with no associated Issue.

Add an explanatory comment on protobuf version pinning. Somewhat obvious with a moment's thought, but no harm in being clear about it.

Motivated by: https://github.com/cylc/cylc-flow/issues/4796#issuecomment-1088365632

### CODE REVIEW

**Comment by hjoliver** [comment]
Aside: we should add a note about this to the installation docs, because it comes close to a requirement to use virtual environments for Cylc, if protobuf is installed on the system for other reasons. (right?)

**Comment by hjoliver** [comment]
(Cancelled tests - not needed for this!)

**Comment by hjoliver** [comment]
Also, from the linked issue:

> With a bit of reading into protobuf releases we could potentially consider loosening this to protobuf==3.* depending on our confidence in back-compatibility between minor releases.

This sounds a bit difficult and dangerous. Perhaps we could just comment (in installation docs) that users can try this themselves if not using a venv, and see if it works.

**Review by oliver-sanders — APPROVED** [review]

File: conda-environment.yml [hunk 1]
```diff
@@ -9,7 +9,12 @@ dependencies:
   - graphene >=2.1,<3
   - graphviz  # for static graphing
   - jinja2 ==2.11.0,<2.12
+<<<<<<< protobuf-version-comment
+  - metomi-isodatetime >=1!2.0.2, <1!2.1.0
+  # Constrain protobuf version for compatible Scheduler-UIS comms across hosts
+=======
   - metomi-isodatetime >=1!3.0.0, <1!3.1.0
+>>>>>>> master
```
Comment (hjoliver) [inline conda-environment.yml:17]:
```suggestion
  - metomi-isodatetime >=1!3.0.0, <1!3.1.0
  # Constrain protobuf version for compatible Scheduler-UIS comms across hosts
```

**Pull request merged**